internal/adapter/persistentvolume.go

```diff
@@ -0,0 +1,180 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/volume"
+	k2dtypes "github.com/portainer/k2d/internal/adapter/types"
+	"github.com/portainer/k2d/internal/k8s"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+func (adapter *KubeDockerAdapter) GetPersistentVolume(ctx context.Context, persistentVolumeName string) (*corev1.PersistentVolume, error) {
+	labelFilter := filters.NewArgs()
+	labelFilter.Add("name", persistentVolumeName)
+	labelFilter.Add("label", fmt.Sprintf("%s=%s", k2dtypes.PersistentVolumeLabelKey, persistentVolumeName))
+
+	volumeList, err := adapter.cli.VolumeList(ctx, volume.ListOptions{Filters: labelFilter})
```

Comment:
Why not inspecting a volume directly instead of using List+Filters?

Reply:
I can switch it to volume inspect, but filters can't be used as it looks with the name defined. I will switch it in any case